common/code/ManagedProtocolSupport/TestAris2Commands/Tests.cs

```diff
@@ -0,0 +1,403 @@
+                {
+                    return Fail("Timed out: " + message);
+                }
+            }
+        }
+
+        private enum VerifyResult { Indeterminate, Success, Failure };
+
+        private static TestResult VerifySeries(
+            FrameListener frameListener,
+            Func<ArisFrameHeader, (bool continueTest, VerifyResult result)> evaluateTest,
+            string message, TimeSpan timeout)
+        {
+            var result = VerifyResult.Indeterminate;
+
+            using (var signal = new ManualResetEventSlim(false))
+            using (var sub = frameListener.FrameHeaders.Subscribe(hdr =>
+            {
+                // success || eval because we might get frames past the first success.
+                bool continueTest;
+                (continueTest, result) = evaluateTest(hdr);
+
+                if (!continueTest)
+                {
+                    Debug.Assert(result == VerifyResult.Success || result == VerifyResult.Failure);
+                    signal.Set();
+                }
+            }))
+            {
+                if (signal.Wait(timeout))
+                {
+                    if (result == VerifyResult.Success)
+                    {
+                        return Pass(message);
+                    }
+                    else
+                    {
+                        return Fail(message);
+                    }
+                }
+                else
+                {
+                    return Fail("Timed out: " + message);
+                }
+            }
+        }
+
+        private static (bool, T) GetAValue<T>(FrameListener frameListener, Func<ArisFrameHeader, T> map, TimeSpan timeout,
+            bool addSlightDelay = true)
+        {
+            // This assumes that the first packet received will be updated.
+            // This in fact relies on the update being applied very quickly,
+            // which may not always happen. So, give it a slight delay.
+
+            if (addSlightDelay)
+            {
+                Thread.Sleep(TimeSpan.FromSeconds(0.2));
+            }
+
+            bool success = false;
+            T value = default(T);
+
+            using (var signal = new ManualResetEventSlim(false))
+            using (var sub = frameListener.FrameHeaders.Take(1).Subscribe(hdr =>
+            {
+                value = map(hdr);
+                signal.Set();
+            }))
+            {
+                success = signal.Wait(timeout);
+                return (success, value);
+            }
+        }
+
+        private static TestResult VerifyChanging<T>(FrameListener frameListener, Func<ArisFrameHeader, T> map, string message,
+            TimeSpan timeout)
+        {
+            // Don't delay getting the initial value.
+            var (gotInitial, initialValue) = GetAValue(frameListener, map, timeout, addSlightDelay: false);
+            if (gotInitial)
+            {
+                return VerifySeries(frameListener,
```

Comment:
Indicate that `true` in the following lambda means "continue series" and `false` means "series complete".
All the `map` arg does is extract a value from the frame header for comparison with `initialValue` but you call it twice and never use `newValue`.
That all took me a while to figure out probably because I'm not used to reading functional code like this in C#.